src/convertToObject.js

```diff
@@ -6,7 +6,15 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  return sourceString.split(';').reduce((acc, prop) => {
+    const [key, value] = prop.split(':').map((str) => str.trim());
```

Comment:
[CODE KNOWLEDGE] - Don't need to make a copy before each operation. In this line, `map` is used correctly to transform each part of the split property, but be aware that `map` does not mutate the original array, so there's no need for concern about unintended side effects.